src/ElectronNET.API/Converter/TitleBarOverlayConverter.cs

```diff
@@ -1,43 +1,46 @@
-﻿using ElectronNET.API.Entities;
-using Newtonsoft.Json;
+using ElectronNET.API.Entities;
 using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
 
 namespace ElectronNET.Converter;
 
 public class TitleBarOverlayConverter : JsonConverter<TitleBarOverlay>
 {
-    public override TitleBarOverlay ReadJson(JsonReader reader, Type objectType, TitleBarOverlay existingValue, bool hasExistingValue, JsonSerializer serializer)
+    public override TitleBarOverlay Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
     {
-        if (reader.TokenType == JsonToken.Boolean)
+        if (reader.TokenType == JsonTokenType.True || reader.TokenType == JsonTokenType.False)
         {
-            return (bool)reader.Value;
+            return (bool)reader.GetBoolean();
         }
-        else if (reader.TokenType == JsonToken.StartObject)
+        else if (reader.TokenType == JsonTokenType.StartObject)
         {
-            return serializer.Deserialize<TitleBarOverlay>(reader);
+            using var doc = JsonDocument.ParseValue(ref reader);
+            return doc.RootElement.Deserialize<TitleBarOverlay>(API.Serialization.ElectronJson.Options);
         }
         else
         {
-            throw new JsonSerializationException("Invalid value for TitleBarOverlay. Expected true, false, or an object.");
+            throw new JsonException("Invalid value for TitleBarOverlay. Expected boolean or an object.");
         }
     }
 
-    public override void WriteJson(JsonWriter writer, TitleBarOverlay value, JsonSerializer serializer)
+    public override void Write(Utf8JsonWriter writer, TitleBarOverlay value, JsonSerializerOptions options)
     {
         if (value is null)
         {
-            writer.WriteUndefined();
+            writer.WriteNullValue();
```

Comment:
See other comment - we need a way to preserve this meaning. `undefined` was written for a reason. Writing `null` will change semantics.